library/Rules/BIC.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Respect\Validation\Rules;
+
+use malkusch\bav\BAV;
+use Respect\Validation\Exceptions\ComponentException;
+
+/**
+ * Validates a BIC (Bank Identifier Code).
+ * 
+ * Currently only German BIC validation is supported (country code "de").
+ * The German validator depends on the composer package malkusch/bav.
+ * Note: It is not recommended to use this validator with BAV's default
+ * configuration. Use a configuration with one of the following
+ * DataBackendContainer implementations:
+ * PDODataBackendContainer or DoctrineBackendContainer.
+ * 
+ * @author Markus Malkusch <markus@malkusch.de>
+ * @see BAV::isValidBIC()
+ * @see \malkusch\bav\Configuration
+ * @see \malkusch\bav\ConfigurationRegistry::setConfiguration()
+ */
+class BIC extends AbstractRule
+{
+    /**
+     * @var String $countryCode The ISO 639-1 country code.
+     */
+    private $countryCode;
+    
+    /**
+     * Sets the country code.
+     * 
+     * The country code is not case sensitive.
+     * 
+     * @param string $countryCode The ISO 639-1 country code.
+     */
+    public function __construct($countryCode)
+    {
+        parent::__construct();
+        $this->countryCode = $countryCode;
+    }
+    
+    public function validate($input)
+    {
+        switch (strtolower($this->countryCode)) {
```

Comment:
C'mon ThIS Is pHP. NoBoDy caRes aBout CaSe. I refered in the docblock to ISO 639-1 which I found e.g. here in lower case: http://www.loc.gov/standards/iso639-2/php/English_list.php The standard itself is behind a paywall.

The code above is BTW case insensitive. Is it a respect convention that a country code has to be in upper case? I'm just curious. I'll change that code to fit your conventions.